go/vt/vttablet/tabletserver/schema/historian_test.go

```diff
@@ -173,3 +174,77 @@ func TestHistorian(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, exp3, fmt.Sprintf("%v", tab))
 }
+
+func TestHistorianPurgeOldSchemas(t *testing.T) {
+	schemaVersionMaxAgeSeconds := 3600 // 1 hour
+	se, db, cancel := getTestSchemaEngine(t, int64(schemaVersionMaxAgeSeconds))
+	defer cancel()
+
+	gtidPrefix := "MySQL56/7b04699f-f5e9-11e9-bf88-9cb6d089e1c3:"
+	gtid1 := gtidPrefix + "1-10"
+	ddl1 := "create table tracker_test (id int)"
+	// create the first record 1 day ago so it gets purged from memory
+	ts1 := time.Now().Add(time.Duration(-24) * time.Hour)
+	_, _, _ = ddl1, ts1, db
+	se.EnableHistorian(true)
+	_, err := se.GetTableForPos(sqlparser.NewIdentifierCS("t1"), gtid1)
+	require.Equal(t, "table t1 not found in vttablet schema", err.Error())
+	var blob1 string
+
+	fields := []*querypb.Field{{
+		Name: "id",
+		Type: sqltypes.Int32,
+	}, {
+		Name: "pos",
+		Type: sqltypes.VarBinary,
+	}, {
+		Name: "ddl",
+		Type: sqltypes.VarBinary,
+	}, {
+		Name: "time_updated",
+		Type: sqltypes.Int32,
+	}, {
+		Name: "schemax",
+		Type: sqltypes.Blob,
+	}}
+
+	table := getTable("t1", []string{"id1", "id2"}, []querypb.Type{querypb.Type_INT32, querypb.Type_INT32}, []int64{0})
+	tables := make(map[string]*binlogdatapb.MinimalTable)
+	tables["t1"] = table
+	blob1 = getDbSchemaBlob(t, tables)
+	db.AddQueryPattern(".*", &sqltypes.Result{
```

Comment:
It's usually better to  be a bit more specific with the pattern for which you expect a given result. We use the specific query below. Any reason not to here as well?

Reply:
I meant to go back and add a more specific regex pattern, but forgot. I'll add that!